File: crescer/las-vegas/api/src/main/java/br/com/cwi/crescer/api/security/controller/LoginController.java
package br.com.cwi.crescer.api.security.controller;

import br.com.cwi.crescer.api.controller.response.LoginResponse;
import br.com.cwi.crescer.api.security.model.UserAccount;
import br.com.cwi.crescer.api.security.repository.UserAccountRepository;
import br.com.cwi.crescer.api.security.service.UserAccountAuthenticatedService;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

@RestController
@RequestMapping("/login")
public class LoginController {

    @Autowired
    UserAccountAuthenticatedService userAccountAuthenticatedService;

    @Autowired
    UserAccountRepository userAccountRepository;

    @PostMapping
    public LoginResponse login() {
        UserAccount user = userAccountAuthenticatedService.get();
        return LoginResponse.builder()
                .permission(user.getPermission())
                .usuarioId(user.getId())
                .build();
    }
}

File: back/springSecurity/src/main/kotlin/com/springSecurity/controller/LoginController.kt
package com.springSecurity.controller

import com.springSecurity.controller.dto.LoginRequest
import com.springSecurity.security.GenerateTokenService
import org.springframework.http.ResponseEntity
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.RequestBody
import org.springframework.web.bind.annotation.RestController

@RestController
class LoginController(
    private val generateTokenService: GenerateTokenService
) {
    @GetMapping("/login")
    fun login(@RequestBody loginRequest: LoginRequest): ResponseEntity<String> =
        ResponseEntity.ok(generateTokenService.generate(loginRequest))
}